### PR TITLE
[OSCI][CLEAN] Audit unused dependencies in OUI

### PR DESCRIPTION
**Pull request by BigSamu**
### Description
Research was done to clean dependencies not used in OUI. Review finished with actions detailed below, but further work has to be done after resolving issue #594 

Actions Performed

1. Following runtime dependencies or devDependencies were removed:

```json
"@types/resize-observer-browser": "^0.1.7",
"rehype-raw": "^5.0.0",
"rehype-stringify": "^8.0.0"
"moment-timezone": "^0.5.41",
"eslint-plugin-jest": "^24.1.0",
```

2.  Following dependencies below were found deprecated and issue #1130  was created to address this.

```json
"@babel/plugin-proposal-class-properties": "^7.10.4",
"@babel/plugin-proposal-object-rest-spread": "^7.11.0",
```
3. Following dependencies below, were reclassified as devDependencies instead of runtime dependencies

```json
"@types/chroma-js": "^2.4.0",
"@types/lodash": "4.14.192",
"@types/numeral": "^2.0.2",
"@types/react-beautiful-dnd": "^13.1.3",
"@types/react-input-autosize": "^2.2.1",
"@types/react-virtualized-auto-sizer": "^1.0.1",
"@types/react-window": "^1.8.5",
"@types/refractor": "^3.0.0",
```

### Issues Resolved
Issue #595 

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md
Comment (BSFishy) [inline CHANGELOG.md:40]:
```suggestion
- Clean up unused runtime and dev dependencies ([#1135](https://github.com/opensearch-project/oui/pull/1135))
```

**Review by BSFishy — COMMENTED** [review]
I want to hold off on approving this until we have integration testing with OSD, just to be as safe as possible

**Review by JohnathonBowers — COMMENTED** [review]
I checked out this PR on my local machine. I was able to get the OUI site to run without issue. Also, I ran `yarn build`, `yarn build-docs`, and `yarn test`, and everything ran successfully. The suggested changes look good to me, although I know that, as @BSFishy said, we still need to wait for the integration tests to be functional.

**Comment by BigSamu** [comment]
> I want to hold off on approving this until we have integration testing with OSD, just to be as safe as possible

Yeah, sure @BSFishy. Make sense!

**Comment by Willie-The-Lord** [comment]
> I want to hold off on approving this until we have integration testing with OSD, just to be as safe as possible

Yeah I agree!

**Comment by joshuarrrr** [comment]
@BigSamu We did update this with `main`, but tests are still failing. Can you take a shot at resolving the conflicts and pushing so we can see the test failures after merge?

**Comment by BSFishy** [comment]
We should merge main into this branch for integ testing with OSD. I would do it myself, but I don't see the button

**Comment by BigSamu** [comment]
> We should merge main into this branch for integ testing with OSD. I would do it myself, but I don't see the button

Hi Matt, what do you mean you don't see the button? Do I need to do something from my side?

**Comment by BigSamu** [comment]
Hi @BSFishy,

Can we retake this Pull Request? What else is required for the merge? I recently updated it to align with the `main` branch.

**Comment by BSFishy** [comment]
> Can we retake this Pull Request? What else is required for the merge? I recently updated it to align with the `main` branch.

It looks like integration with OSD is failing: https://github.com/opensearch-project/oui/actions/runs/7378461865/job/20073761573?pr=1135#step:9:207

We would want to fix that before merging.

**Comment by BigSamu** [comment]
> > Can we retake this Pull Request? What else is required for the merge? I recently updated it to align with the `main` branch.
> 
> It looks like integration with OSD is failing: https://github.com/opensearch-project/oui/actions/runs/7378461865/job/20073761573?pr=1135#step:9:207
> 
> We would want to fix that before merging.

Yeah, I saw that. But is this failure related to something other than our PR? I understand those checks with integration tests to OSD were an issue on your side, didn't it?

**Comment by BigSamu** [comment]
> > > Can we retake this Pull Request? What else is required for the merge? I recently updated it to align with the `main` branch.
> > 
> > 
> > It looks like integration with OSD is failing: https://github.com/opensearch-project/oui/actions/runs/7378461865/job/20073761573?pr=1135#step:9:207
> > We would want to fix that before merging.
> 
> Yeah, I saw that. But is this failure related to something other than our PR? I understand those checks with integration tests to OSD were an issue on your side, didn't it?

@BSFishy,

At last, I found the problem. The typescript dependencies listed below through an error in the OSD integration tests when they are installed as `devDependenceis`

```js
"@types/chroma-js": "^2.4.0",
"@types/react-beautiful-dnd": "^13.1.3",
"@types/react-input-autosize": "^2.2.1",
"@types/react-window": "^1.8.5",
```

This is weird. Should not these dependencies only be used in development? Why are they required on runtime? Maybe there is a problem with the build process during the OSD integration tests.

**Comment by BSFishy** [comment]
> This is weird. Should not these dependencies only be used in development? Why are they required on runtime? Maybe there is a problem with the build process during the OSD integration tests.

Yeah this is super weird to me too. I know OSD has some non-standard bootstrap procedures which may be causing these issues. I think @AMoo-Miki might have some more insights into how OSD bootstrap works and why this may be an issue

**Comment by BigSamu** [comment]
> > This is weird. Should not these dependencies only be used in development? Why are they required on runtime? Maybe there is a problem with the build process during the OSD integration tests.
> 
> Yeah this is super weird to me too. I know OSD has some non-standard bootstrap procedures which may be causing these issues. I think @AMoo-Miki might have some more insights into how OSD bootstrap works and why this may be an issue

So, as you said, @BSFishy, @AMoo-Miki reply to this inquiry in our internal chats. I added his response below to update everyone. So I think we are good to go now, or not?

From Miki:

_"OSD and plugins don’t ONLY use OUI’s disted artifacts and hence, OSD type-checks OUI during build. If OUI doesn’t have its own deps installed when placed in OSD, the type-check and build will fail. So, either OUI will have to redist those types in its own d-ts, or it has to rely on the defs it finds. They chose to use the defs it finds and to avoid conflicts with any types that OSD might have, they just added these as regular deps"_

**Review by BSFishy — APPROVED** [review]
It may be possible to move these `@types/` dependencies into dev dependencies, but this PR fulfills its goals.

If I'm not mistaken, the issue is that OUI exposes types from these dependencies in its own public API. This means that OSD will need to be able to reference these types as well. So if we're able to trick Typescript into inlining these types in the `.d.ts` file, we could presumably get rid of these dependencies. This could be taken up in a followup issue

File: package.json
Comment (AMoo-Miki) [inline package.json:215]: `moment-timezone` enriches `moment` and needs to stay.

Comment (BigSamu) [reply package.json:215]: @AMoo-Miki dependency added back!

File: CHANGELOG.md
Comment (AMoo-Miki) [inline CHANGELOG.md:24]: This is not part of this PR.

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]

File: package.json
Comment (joshuarrrr) [reply package.json:215]: I'd keep the `^` that was there before

File: src-docs/src/views/flex/flex_basis_string.tsx
Comment (joshuarrrr) [inline src-docs/src/views/flex/flex_basis_string.tsx:1]: @BigSamu It looks like this PR branch got mixed up with the flex changes from another branch. This PR should just have 3 files, right (package.json, yarn.lock, and CHANGELOG.md)?

**Review by joshuarrrr — CHANGES_REQUESTED** [review]
It looks like this branch has some of the flex related changes from another PR.

File: src-docs/src/views/flex/flex_basis_string.tsx
Comment (BigSamu) [reply src-docs/src/views/flex/flex_basis_string.tsx:1]: Ouuuu, you're right!!! My god. Just fixed! Thanks for the heads up @joshuarrrr

**Review by joshuarrrr — APPROVED** [review]

**Pull request merged**
